modules/AlphaWalletFoundation/AlphaWalletFoundation/Transactions/BlockscoutApiNetworking.swift

```diff
@@ -0,0 +1,318 @@
+//
+//  BlockscoutApiNetworking.swift
+//  AlphaWalletFoundation
+//
+//  Created by Vladyslav Shepitko on 09.05.2023.
+//
+
+import Foundation
+import SwiftyJSON
+import Combine
+import AlphaWalletCore
+import BigInt
+import AlphaWalletLogger
+import Alamofire
+
+class BlockscoutApiNetworking: ApiNetworking {
+    private let server: RPCServer
+    private let transporter: ApiTransporter
+    private let transactionBuilder: TransactionBuilder
+    private let apiKey: String?
+    private let baseUrl: URL
+
+    init(server: RPCServer,
+         transporter: ApiTransporter,
+         transactionBuilder: TransactionBuilder,
+         apiKey: String?,
+         baseUrl: URL) {
+
+        self.baseUrl = baseUrl
+        self.apiKey = apiKey
+        self.transactionBuilder = transactionBuilder
+        self.transporter = transporter
+        self.server = server
+    }
+
+    func erc20TokenInteractions(walletAddress: AlphaWallet.Address,
+                                startBlock: Int?) -> AnyPublisher<UniqueNonEmptyContracts, PromiseError> {
+
+        let request = Request(
+            baseUrl: baseUrl,
+            startBlock: startBlock,
+            apiKey: apiKey,
+            walletAddress: walletAddress,
+            action: .tokentx)
+
+        return transporter
+            .dataTaskPublisher(request)
+            .handleEvents(receiveOutput: { [server] in EtherscanCompatibleApiNetworking.log(response: $0, server: server) })
+            .tryMap { UniqueNonEmptyContracts(json: try JSON(data: $0.data), tokenType: .erc20) }
+            .mapError { PromiseError.some(error: $0) }
+            .print("xxx.blockscout.normalTransactions.server: \(server)")
+            .eraseToAnyPublisher()
+    }
+
+    func erc721TokenInteractions(walletAddress: AlphaWallet.Address,
+                                 startBlock: Int?) -> AnyPublisher<UniqueNonEmptyContracts, PromiseError> {
+
+        switch server {
```

Comment:
Is this switch for logging the same for all cases in both `EtherscanCompatibleApiNetworking` and `BlockscoutApiNetworking`? Maybe useful to extract it out into an extension of `ApiNetworking`? Should be useful to keep it around and still log it with `debugLog()` and even crash with `precondition()`

Reply:
its a bit different for eip 1155